packages/client-config/src/client-config-writer/client_config_converter.ts

```diff
@@ -144,6 +144,65 @@ export class ClientConfigConverter {
 
       mobileConfig.geo = geoConfig;
     }
+
+    if (clientConfig.Analytics) {
+      mobileConfig.analytics = {
+        plugins: {
+          awsPinpointAnalyticsPlugin: {
+            pinpointAnalytics: {
+              region: clientConfig.Analytics.AWSPinpoint.region,
+              appId: clientConfig.Analytics.AWSPinpoint.appId,
+            },
+            pinpointTargeting: {
+              region: clientConfig.Analytics.AWSPinpoint.region,
+            },
+          },
+        },
+      };
+    }
+
+    if (clientConfig.Notifications) {
+      // APNS and FCM are mapped to the same awsPinpointPushNotificationsPlugin
+      // Throw if they're both present but defined differently
+      // as this is ambiguous situation
+      const fcm = clientConfig.Notifications.FCM;
+      const apns = clientConfig.Notifications.APNS;
+      if (
+        fcm &&
+        apns &&
+        (fcm.AWSPinpoint.appId !== apns.AWSPinpoint.appId ||
+          fcm.AWSPinpoint.region !== apns.AWSPinpoint.region)
+      ) {
+        throw new Error(
+          'Cannot convert client config to mobile config if both FCM and APNS are defined with different AWS Pinpoint instance'
+        );
+      }
```

Comment:
This is one gotcha that I encountered.
The alternative would be to expand `ClientConfigFormat` and instead of `mobile` have android and ios separately.

@josefaidt Can we make this tradeoff and pursue better fix with config unification ?

Reply:
> btw. have you ever seen gen1 app that would use different pinpoint instance for fcm and apns ? it seems that gen1 is just enabling channels there for one instance...

No, but it does seem like you _could_ wire the backend together to support more than one